File: use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/domain/Timeslot.java
package org.acme.schooltimetabling.domain;

import java.time.DayOfWeek;
import java.time.LocalTime;

import ai.timefold.solver.core.api.domain.common.PlanningId;

/**
 * A teaching slot on a particular day and time range. Lessons are assigned to a
 * {@link Timeslot} by the solver.
 */
public class Timeslot {

    @PlanningId
    private String id;

    private DayOfWeek dayOfWeek;
    private LocalTime startTime;
    private LocalTime endTime;

    public Timeslot() {
    }

    public Timeslot(String id, DayOfWeek dayOfWeek, LocalTime startTime, LocalTime endTime) {
        this.id = id;
        this.dayOfWeek = dayOfWeek;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    public String getId() {
        return id;
    }

    public DayOfWeek getDayOfWeek() {
        return dayOfWeek;
    }

    public LocalTime getStartTime() {
        return startTime;
    }

    public LocalTime getEndTime() {
        return endTime;
    }

    @Override
    public String toString() {
        return dayOfWeek + " " + startTime;
    }
}

File: technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/domain/Timeslot.kt
package org.acme.kotlin.schooltimetabling.domain

import ai.timefold.solver.core.api.domain.lookup.PlanningId
import com.fasterxml.jackson.annotation.JsonIdentityInfo
import com.fasterxml.jackson.annotation.ObjectIdGenerators
import java.time.DayOfWeek
import java.time.LocalTime

@JsonIdentityInfo(
    scope = Timeslot::class,
    generator = ObjectIdGenerators.PropertyGenerator::class,
    property = "id"
)
data class Timeslot(
    @PlanningId
    val id: String,
    val dayOfWeek: DayOfWeek,
    val startTime: LocalTime,
    val endTime: LocalTime = startTime.plusMinutes(50)) {

    override fun toString(): String = "$dayOfWeek $startTime"

}
